gcsfs/core.py

```diff
@@ -296,13 +329,47 @@ def _call(self, method, path, *args, **kwargs):
         meth = getattr(requests, method)
         if args:
             path = path.format(*[quote_plus(p) for p in args])
-        r = meth(self.base + path, headers=self.header, params=kwargs,
-                 json=json)
+        for retry in range(RETRY_COUNT):
+            try:
+                time.sleep(2**retry-1)
+                r = meth(self.base + path, headers=self.header, params=kwargs,
+                         json=json)
+                validate_response(r, path)
+                break
+            except RuntimeError as e:
```

Comment:
Is there anything fundamentally different about the two fail blocks, or can they be combined?

Reply:
In the case of the RuntimeError we have to extract the HTTP error code.  There isn't much code beyond that (especially once we remove the logging and debug code I forgot to remove).

Reply:
So runtime error is produced by requests?
If it's the one raised in validate_response (which I think it must be), then r already is defined and will always have the status_code attribute available - to avoid having to parse the contents.
If parsing is the only way to do this (hope not!), then it would be better if the parsing code were in the util module. It could also do with tests of its own.